### PR TITLE
Add connection options

### PR DESCRIPTION
**Pull request by m-cat**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -12,14 +13,49 @@ function defaultOptions(endpointPath) {
...
+ * @param {string} [config.customUserAgent=""] - Custom user agent header to set.
+ * @param {Object} [config.data=null] - Data to send in a POST.
+ * @param {string} [config.endpointPath=""] - The relative URL path of the portal endpoint to contact.
+ * @param {string} [config.extraPath=""] - Extra path element to append to the URL.
+ * @param {Function} [config.onUploadProgress] - Optional callback to track progress.
+ * @param {Object} [config.params={}] - Query parameters to include in the URl.
+ * @param {string} [config.responseType=""] - The type of data that the server will respond with.
+ */
+function executeRequest(config) {
+  let url = makeUrl(config.portalUrl, config.endpointPath, config.extraPath ? config.extraPath : "");
```
Comment (kwypchlo) [inline src/utils.js:36]:
```suggestion
  const url = makeUrl(config.portalUrl, config.endpointPath, config.extraPath ? config.extraPath : "");
```

File: src/utils.js [hunk 2]
```diff
@@ -12,14 +13,49 @@ function defaultOptions(endpointPath) {
...
+
+  return axios({
+    url: url,
```
Comment (kwypchlo) [inline src/utils.js:39]:
```suggestion
    url,
```

File: src/utils.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 "use strict";
 
+const axios = require("axios");
 const p = require("path");
```
Comment (kwypchlo) [inline src/utils.js:4]: use `path` for that, abbreviating to one letter variables reduces readability

**Review by kwypchlo — CHANGES_REQUESTED** [review]

File: src/utils.js (hunk 3 above)
Comment (m-cat) [reply src/utils.js:4]: k, blame points to you though 😉

**Review by kwypchlo — APPROVED** [review]

**Pull request merged**